### PR TITLE
[fix-#14174] ```sudo.enable = false ```  not work

### PR DESCRIPTION
**Pull request by fuchanghai**
## Purpose of the pull request
this close : #14174 
- #14174 

## Verify this pull request

This pull request is code cleanup without any test coverage.

### CODE REVIEW

**Comment by fuchanghai** [comment]
@zhuangchong @zhongjiajie PTAL

**Comment by zhuangchong** [comment]
You can compare the changes of this pr #13860

**Comment by fuchanghai** [comment]
> You can compare the changes of this pr #13860
if not add this judge ,The task has failed when it is submitted to the worker.
image

File: dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/processor/TaskDispatchProcessor.java [hunk 1]
```diff
@@ -131,7 +132,7 @@ public void process(Channel channel, Command command) {
                 }
 
                 // check if the OS user exists
-                if (!OSUtils.getUserList().contains(taskExecutionContext.getTenantCode())) {
+                if (PropertyUtils.getBoolean(Constants.SUDO_ENABLE, true) && !OSUtils.getUserList().contains(taskExecutionContext.getTenantCode())) {
```
Comment (caishunfeng) [inline dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/processor/TaskDispatchProcessor.java:135]: I don't think we should add this sudo judgment here.
It's need to check tenant exist if we specify it in workflow, independent of whether or not you can sudo tenant.
WDYT?

**Comment by caishunfeng** [comment]
> SonarCloud Quality Gate failed.    [Quality Gate failed](https://sonarcloud.io/dashboard?id=apache-dolphinscheduler&pullRequest=14176)
> 
> [Bug](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=14176&resolved=false&types=BUG) [D](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=14176&resolved=false&types=BUG) [25 Bugs](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=14176&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=14176&resolved=false&types=VULNERABILITY) [C](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=14176&resolved=false&types=VULNERABILITY) [14 Vulnerabilities](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=14176&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=14176&resolved=false&types=SECURITY_HOTSPOT) [E](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=14176&resolved=false&types=SECURITY_HOTSPOT) [4 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=14176&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=14176&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=14176&resolved=false&types=CODE_SMELL) [805 Code Smells](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=14176&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=14176&metric=new_coverage&view=list) [35.5% Coverage](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=14176&metric=new_coverage&view=list) [](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=14176&metric=new_duplicated_lines_density&view=list) [5.8% Duplication](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=14176&metric=new_duplicated_lines_density&view=list)

BTW, the sonar result seems not right. :eyes:

**Pull request closed without merging**